File: kotlin/src/main/kotlin/cl/payments/adapters/AdapterFactory.kt
package cl.payments.adapters

import cl.payments.adapters.base.PaymentAdapter
import cl.payments.adapters.webpay.WebpayAdapter
import cl.payments.adapters.getnet.GetnetAdapter
import cl.payments.types.PaymentProvider
import cl.payments.types.SDKConfig

// Factory para crear adaptadores de pago
object AdapterFactory {
    // Crear adaptador según el proveedor configurado
    fun create(config: SDKConfig): PaymentAdapter {
        return when (config.provider) {
            PaymentProvider.WEBPAY -> WebpayAdapter(config)
            PaymentProvider.GETNET -> GetnetAdapter(config)
        }
    }
}

File: src/adapters/AdapterFactory.ts
import { SDKConfig, ValidationError } from '../types';
import { PaymentAdapter } from './base/PaymentAdapter';
import { WebpayAdapter } from './webpay/WebpayAdapter';
import { GetnetAdapter } from './getnet/GetnetAdapter';

// Factory para crear adaptadores de pago
export class AdapterFactory {
  // Crear adaptador según el proveedor configurado
  static create(config: SDKConfig): PaymentAdapter {
    switch (config.provider) {
      case 'webpay':
        return new WebpayAdapter(config);
      
      case 'getnet':
        return new GetnetAdapter(config);
      
      default:
        throw new ValidationError(
          `Unsupported payment provider: ${config.provider}`
        );
    }
  }
}